src/m365/planner/commands/task/task-remove.ts

```diff
@@ -0,0 +1,256 @@
+      })
+      .then((buckets) => {
+        const filteredBuckets = buckets.value.filter(b => bucketName!.toLocaleLowerCase() === b.name!.toLocaleLowerCase());
+        
+        if (!filteredBuckets.length) {
+          return Promise.reject(`The specified bucket ${bucketName} does not exist`);
+        }
+
+        if (filteredBuckets.length > 1) {
+          return Promise.reject(`Multiple buckets with name ${bucketName} found: Please disambiguate:${os.EOL}${filteredBuckets.map(f => `- ${f.id}`).join(os.EOL)}`);
+        }
+
+        return Promise.resolve(filteredBuckets[0].id!);
+      });
+  }
+
+  private getPlanId(options: Options): Promise<string> {
+    const { planId, planName } = options;
+
+    if (planId) {
+      return Promise.resolve(planId);
+    }
+
+    return this
+      .getGroupId(options)
+      .then(groupId => planner.getPlanByName(planName!, groupId))
+      .then(plan => plan.id!);
+  }
+
+  private getGroupId(options: Options): Promise<string> {
+    const { ownerGroupId, ownerGroupName } = options;
+
+    if (ownerGroupId) {
+      return Promise.resolve(ownerGroupId);
+    }
+
+    const requestOptions: AxiosRequestConfig = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(ownerGroupName!)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: Group[] }>(requestOptions)
+      .then(response => {
+        if (!response.value.length) {
+          return Promise.reject(`The specified owner group ${ownerGroupName} does not exist`);
+        }
+
+        if (response.value.length > 1) {
+          return Promise.reject(`Multiple owner groups with name ${ownerGroupName} found: ${response.value.map(x => x.id)}`);
+        }
+
+        return Promise.resolve(response.value[0].id!);
+      });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      { option: '-i, --id [id]' },
+      { option: '-t, --title [title]' },
+      { option: '--bucketId [bucketId]' },
+      { option: '--bucketName [bucketName]' },
+      { option: '--planId [planId]' },
+      { option: '--planName [planName]' },
+      { option: '--ownerGroupId [ownerGroupId]' },
+      { option: '--ownerGroupName [ownerGroupName]' },
+      { option: '--confirm' }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+  
+  public validate(args: CommandArgs): boolean | string {
+    if (args.options.id) {
+      if (args.options.bucketId || args.options.bucketName || args.options.planId || args.options.planName || args.options.ownerGroupId || args.options.ownerGroupName) {
+        return 'Don\'t specify bucketId,bucketName, planId, planName, ownerGroupId or ownerGroupName when using id';
+      }
+      if (args.options.title) {
+        return 'Specify either id or title';
```

Comment:
For this validation: (id or title required) you can use the new functionality: optionSets. Check bucket-remove.ts for its implementation.